app/gameengine.py

```diff
@@ -4780,6 +4850,54 @@ def validate_main_step_place_holomem(self, player_id:str, action_data:dict):
 
         return True
 
+    def validate_main_step_attached_action(self, player_id: str, action_data: dict) -> bool:
+        if not self.validate_decision_base(player_id, action_data, DecisionType.DecisionMainStep, GameAction.MainStepAttachedActionFields):
+            return False
+
+        player = self.get_player(player_id)
+
+        # Validate that the support card exists
+        support_id = action_data["support_id"]
```

Comment:
as I mentioned earlier, if this becomes a generic special actions handler, the caller will just send the index and you'll make sure GetAvailableSpecialActions has an item with that index.